game/src/sandbox/dashboards/trip_problems.rs

```diff
@@ -220,29 +232,40 @@ impl<X: Copy + PartialOrd + Display, Y: Copy + PartialOrd + Display> Matrix<X, Y
             .unwrap_or(self.buckets_y.len())
             - 1;
         let idx = self.idx(x_idx, y_idx);
-        self.counts[idx] += 1;
+        self.entries[idx].push(id);
     }
 
     fn idx(&self, x: usize, y: usize) -> usize {
         // Row-major
         y * self.buckets_x.len() + x
     }
 
-    fn draw(self, ctx: &mut EventCtx, app: &App, opts: MatrixOptions<X, Y>) -> Widget {
+    fn draw(
+        mut self,
+        ctx: &mut EventCtx,
+        app: &App,
+        opts: MatrixOptions<X, Y>,
+    ) -> (Widget, HashMap<String, Vec<ID>>) {
         let mut grid_batch = GeomBatch::new();
         let mut tooltips = Vec::new();
         let cell_width = opts.total_width / (self.buckets_x.len() as f64);
         let cell_height = opts.total_height / (self.buckets_y.len() as f64);
         let cell = Polygon::rectangle(cell_width, cell_height);
 
-        let max_count = *self.counts.iter().max().unwrap() as f64;
+        let max_count = self.entries.iter().map(|list| list.len()).max().unwrap() as f64;
 
+        let mut mapping = HashMap::new();
         for x in 0..self.buckets_x.len() - 1 {
             for y in 0..self.buckets_y.len() - 1 {
                 let is_first_xbucket = x == 0;
                 let is_last_xbucket = x == self.buckets_x.len() - 2;
                 let is_middle_ybucket = y + 1 == self.buckets_y.len() / 2;
-                let count = self.counts[self.idx(x, y)];
+                let idx = self.idx(x, y);
+                let count = self.entries[idx].len();
+                let bucket_label = format!("{}/{}", opts.title, idx);
```

Comment:
This is the string returned in `Outcome::Clicked`. It's never displayed to the user; it just needs to be unique over the whole panel and match up to the `Vec<TripID>`